src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -22,6 +22,80 @@ class DatasetConverter(pydantic.BaseModel):
         default=None,
     )
 
+    @classmethod
+    @pydantic.validate_call
+    def from_dandiset(
+        cls,
+        dandiset_id: pydantic.DirectoryPath,
+        version_id: str | None = None,
+        api_url: str | None = None,
+        token: str | None = None,
+    ) -> typing_extensions.Self:
+        """
+        Initialize a converter of Dandiset to BIDS format.
+
+        Parameters
+        ----------
+        dandiset_id : directory path
+            The path to the Dandiset directory.
+        archive : str, optional
+            The archive of the Dandiset. Common options are "DANDI" or "EMBER".
+            The default is "DANDI".
+        """
+        import dandi.dandiapi
+
+        client = dandi.dandiapi.DandiAPIClient(api_url=api_url, token=token)
+        dandiset = client.get_dandiset(dandiset_id=dandiset_id, version_id=version_id)
```

Comment:
think about abstracting the source of the listing of nwb files with URLs (consider file:// urls too! we have some `datalad get`ed already) to them, and producing remapping + extra bids files.  Then we could easily make it work on e.g. datalad dandisets with git-annex to produce "bids" branch for all of them!

Reply:
Moved to #46 

This PR is quite enough already with focusing only on purely remote usage through the DANDI API + remfile so I'll leave that abstraction for a future PR (and probably a different instance method as well)